### PR TITLE
Remove dependency on six

### PR DESCRIPTION
**Pull request by bmw**
Fixes https://github.com/certbot/certbot/issues/8494.

You can see the full test suite passing with these changes at https://dev.azure.com/certbot/certbot/_build/results?buildId=3409&view=results.

I left the `six` dependency pinned in `tests/letstest/requirements.txt` and `tools/oldest_constraints.txt` because `six` is still a transitive dependency with our current pinnings.

The extra moving around of imports is due to me using `isort` to help me keep dependencies in sorted order after replacing imports of `six`.

### CODE REVIEW

File: certbot/certbot/tests/util.py [hunk 1]
```diff
@@ -36,6 +26,18 @@
 from certbot.compat import os
 from certbot.display import util as display_util
 
+try:
```
Comment (adferrand) [inline certbot/certbot/tests/util.py:29]: I think this is a consequence of executing `isort` (I got that serveral time). Did you have the intention to move this import there?

File: certbot/tests/cli_test.py [hunk 2]
```diff
@@ -21,6 +16,12 @@
 import certbot.tests.util as test_util
 from certbot.tests.util import TempDirTestCase
 
+try:
```
Comment (adferrand) [inline certbot/tests/cli_test.py:19]: See my comment above.

File: certbot/tests/display/util_test.py [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
-except ImportError: # pragma: no cover
-    from unittest import mock
-import six
 
 from certbot import errors
 from certbot import interfaces
 from certbot.display import util as display_util
 import certbot.tests.util as test_util
 
+try:
```
Comment (adferrand) [inline certbot/tests/display/util_test.py:14]: See my comment above.

File: certbot/tests/log_test.py [hunk 4]
```diff
@@ -19,6 +15,12 @@
 from certbot.compat import os
 from certbot.tests import util as test_util
 
+try:
```
Comment (adferrand) [inline certbot/tests/log_test.py:18]: See my comment above.

File: certbot/tests/main_test.py [hunk 5]
```diff
@@ -39,6 +35,12 @@
 from certbot.plugins import enhancements
 import certbot.tests.util as test_util
 
+try:
```
Comment (adferrand) [inline certbot/tests/main_test.py:38]: See my comment above.

File: certbot/tests/util_test.py [hunk 6]
```diff
@@ -1,20 +1,21 @@
...
+from importlib import reload as reload_module
+import io
 import sys
 import unittest
 
+from certbot import errors
+from certbot.compat import filesystem
+from certbot.compat import os
+import certbot.tests.util as test_util
+
```
Comment (adferrand) [inline certbot/tests/util_test.py:13]: See my comment above.

**Review by adferrand — CHANGES_REQUESTED** [review]
You have not mentionned the `six` dependency in `certbot-auto`, but I think we do not care anymore. Only one question about moves for the `mock` import, and we are good to go!

File: certbot/certbot/tests/util.py (hunk 1 above)
Comment (bmw) [reply certbot/certbot/tests/util.py:29]: You're right that this was due to `isort`. I tried to explain this in my initial post when I said:

> The extra moving around of imports is due to me using `isort` to help me keep dependencies in sorted order after replacing imports of `six`.

I can change them back if you want, however, I personally don't mind the change and think we may as well sort imports as done by `isort` to make it easier for us to use the tool.

Comment (adferrand) [reply certbot/certbot/tests/util.py:29]: In fact, it superiorily annoys me each time I need to manually fix the imports on that matter each time I run isort, so I am fine with you keeping the order as it is in the PR.

Anyway things will be back ti normal once we remove the `mock` dependency and do not need to do a try/except on it anymore.

**Review by adferrand — APPROVED** [review]

File: certbot/certbot/tests/util.py (hunk 1 above)
Comment (bmw) [reply certbot/certbot/tests/util.py:29]: > In fact, it superiorily annoys me each time I need to manually fix the imports on that matter each time I run isort,

That'd annoy me too! In general if there are things on Certbot that are making your life harder, by all means flag/ask about them. Let's not make things harder than they need to be.

**Pull request merged**